### DISCUSSION TITLE
[question] Could you please tell us how to to validate the input parameters( for pass partially ) when using tRPC?

### DISCUSSION THREAD

**Discussion by liaodalin19903** [trpc/trpc#6322] [Q&A]
Hello everyone! We are currently trying to use tRPC combined with TypeORM to update a database table (Person).

We've written the following code:

```typescript
export const personUpdate = publicProcedure.input(
  z.object({
    id: z.number(),
    personName: z.string(),
    desc: z.string()
  })
).mutation(({ input: {id, personName, desc} }) => {
  dataBase.getRepository(Person).createQueryBuilder('person').where('person.id = :id', {id: id}).update({
    personName: personName,
    desc: desc
  })
}) 
```

However, we now have a question.

As you can see, we are using Zod to validate the input parameters here. Since we know that when updating a piece of data, sometimes it's sufficient to only pass in the necessary modified fields, for example, the `Partial<Person>` type.

Could you please tell us how we can optimize this code when using tRPC?

**Comment by OtayNacef** [comment]
Use .partial() Check out the docs https://zod.dev/?id=partial
if there is a required attributes like id, you can use:
z.object({ id: z.number() }).merge(z.object({ personName: z.string(), desc: z.string() }).partial())
